src/nvim/eval/funcs.c

```diff
@@ -1678,7 +1678,13 @@ static void f_delete(typval_T *argvars, typval_T *rettv, FunPtr fptr)
     rettv->vval.v_number = os_rmdir(name) == 0 ? 0 : -1;
   } else if (strcmp(flags, "rf") == 0) {
     // delete a directory recursively
+
+    // Note: delete_recursive() some time fails.
+    // If it is failed, it should retry.
```

Comment:
Why? What if it fails again?

Reply:
Which tests are more flaky without retrying here?

Reply:
I don't know the accurate reason.  But `os_remove()` seems failed in unit tests.
But if retry, it is success.  It is concurrent problem?

Of course, Vim has not the problem.

Reply:
flaky tests are them.

https://github.com/neovim/neovim/pull/16268#issuecomment-972464856